### PR TITLE
job improvements

### LINKED ISSUE #389: Catch Errors in Job.Execute
Bot crashes with error in jobs sometimes

### PR DESCRIPTION
**Pull request by suneettipirneni**
Fixes #389, 

When a job starts it's in the format

`"JOB-(JOB_NAME) started"`

In addition it makes the `name` and `execute` abstract properties in the `Job` class since all of the job subclasses implement them anyways.

### CODE REVIEW

File: src/services/job.service.ts [hunk 1]
```diff
@@ -23,7 +23,12 @@ export class JobService {
       throw new Error(`Job ${job.name} already exists as a running job.`);
     }
     this._runningJobs[job.name] = setInterval(() => {
-      return job.execute(container);
+      try {
+        container.loggerService.info(`${job.constructor.name}-(${job.name}) started.`);
+        job.execute(container);
+      } catch(error) {
+        container.loggerService.error(error);
+      }
```
Comment (tanndlin) [inline src/services/job.service.ts:31]: Follow same flow as plugins
https://github.com/cs-discord-at-ucf/lion/blob/4e8ec75bb9c4bb546a2e31003cdc76d5638b249b/src/app/handlers/command.handler.ts#L154-L164

**Review by tanndlin — APPROVED** [review]
lgtm, whatcha think @c650

**Comment by suneettipirneni** [comment]
This PR is getting stale, could I get another review?

**Review by Armintron — APPROVED** [review]
Looks good to me

**Review by c650 — APPROVED** [review]

**Pull request merged**